### PR TITLE
Linter enabled tests

### PR DESCRIPTION
**Pull request by kevinclark19a**
## Description

Enable linter to run on unit tests found as PR gate, and added error suppression to existing errors.

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: 2.7.pylintrc [hunk 1]
```diff
@@ -8,28 +8,38 @@
 # attribute-defined-outside-init<W0201>: (hi-pri) Used when an instance attribute is defined outside the __init__ method.
 # bad-continuation<C0330>: (needs review) **REMOVED in pylint-2.6.0**
 # bad-indentation<W0311>: (hi-pri) Used when an unexpected number of indentation's tabulations or spaces has been found.
+# bad-open-mode<W1501>: (hi-pri) Python supports: r, w, a[, x] modes with b, +, and U (only with r) options.
```
Comment (narrieta) [inline 2.7.pylintrc:11]: this is a comment for line 9 above

># bad-continuation<C0330>: (needs review) **REMOVED in pylint-2.6.0**

maybe we should disable this one globally (to the entire project)?

File: tests/common/test_cgroupapi.py [hunk 2]
```diff
@@ -65,24 +65,24 @@ def test_cgroups_should_be_supported_only_on_ubuntu_16_and_later(self):
             (['redhat', '7.7.1908', 'Core'], False),
             (['bigip', '15.0.1', 'Final'], False),
             (['gaia', '273.562', 'R80.30'], False),
-            (['debian' '9.1', ''], False),
+            (['debian' '9.1', ''], False), # pylint: disable=implicit-str-concat,implicit-str-concat-in-sequence
```
Comment (narrieta) [inline tests/common/test_cgroupapi.py:68]: Nice! 

This is my bad. I just submitted #1986  fixing this. We can remove this suppression.

File: 2.7.pylintrc (hunk 1 above)
Comment (kevinclark19a) [reply 2.7.pylintrc:11]: We can definitely do that.

Note that python2.6 and python3.4 are stuck on earlier versions that _do_ have this symbol (any of 3.[6,7,8] might also get stuck on current stable, 2.5.3, which also has this symbol), so I'm not sure if that changes things. We could potentially leave this enabled in this file, and disable it for other versions once they update.

File: tests/common/test_cgroupapi.py (hunk 2 above)
Comment (kevinclark19a) [reply tests/common/test_cgroupapi.py]: Done!

File: tests/common/mock_cgroup_commands.py [hunk 3]
```diff
@@ -28,13 +28,16 @@
 #
 # The output comes from an Ubuntu 18 system
 #
-_default_commands = [
+_default_commands = [ # pylint: disable=invalid-name
     (r"^systemctl --version$",
+    # pylint: disable=bad-continuation
```
Comment (larohra) [inline tests/common/mock_cgroup_commands.py:33]: NIT: Remove this maybe, since its globally disabled

File: tests/protocol/test_datacontract.py [hunk 4]
```diff
@@ -20,10 +20,10 @@
 from azurelinuxagent.common.datacontract import get_properties, set_properties, DataContract, DataContractList
 
 
-class SampleDataContract(DataContract):
+class SampleDataContract(DataContract): # pylint: disable=too-few-public-methods
     def __init__(self):
-        self.foo = None
-        self.bar = DataContractList(int)
+        self.foo = None # pylint: disable=blacklisted-name
```
Comment (larohra) [inline tests/protocol/test_datacontract.py:25]: I couldn't find a definition for this, could you add it please! Thanks!

Comment (kevinclark19a) [reply tests/protocol/test_datacontract.py:25]: This error is only reported in python{2.7,3.4}, so I only put the definition in `2.6.pylintrc`.

I'm assuming you looked in `3.6.pylintrc` and couldn't find it?

Comment (larohra) [reply tests/protocol/test_datacontract.py:25]: You mean the `2.7.pylintrc`? I looked at both (for this branch) and couldn't find it. Is it part of some other branch?

Comment (kevinclark19a) [reply tests/protocol/test_datacontract.py:25]: Sorry, yes, typo.

I'm seeing it on line 13 in that file, both on my local branch, and in this PR. Maybe you're not looking at all commits?

Comment (larohra) [reply tests/protocol/test_datacontract.py:25]: Ahh got it, I was in the wrong branch. Thanks for confirming :)

**Pull request merged**